tests/IntegrationTest.php

```diff
@@ -73,4 +74,53 @@ public function gettingEncryptedStuffFromGoogleShouldWork()
         $this->assertTrue($connected);
         $this->assertRegExp('#^HTTP/1\.0#', $response);
     }
+
+    /** @test */
+    public function testSelfSignedRejectsIfVerificationIsEnabled()
```

Comment:
This is strange indeed…

Can you try running the following code to verify this is in line with what PHP's stream handlers do?

```php
file_get_contents('https://self-signed.badssl.com/', false, stream_context_create(array('ssl'=>array('verify_peer'=>false))));
// all okay

file_get_contents('https://self-signed.badssl.com/', false, stream_context_create(array('ssl'=>array('verify_peer'=>true))));
// PHP warning:  file_get_contents(): SSL operation failed with code 1. OpenSSL Error messages:
// error:14090086:SSL routines:SSL3_GET_SERVER_CERTIFICATE:certificate verify failed on line 1
```